tests/storages_tests/rdb_tests/test_storage.py

```diff
@@ -215,3 +217,103 @@ def setup_trials(storage, study_id):
     ) as mock_object:
         assert storage.get_all_trials(study_id) == trials
         assert mock_object.call_count == 1
+
+
+def test_update_trial() -> None:
+    storage = create_test_storage()
+    study_id = storage.create_new_study()
+
+    test_sets = [
+        (
+            {"state": TrialState.COMPLETE, "datetime_complete": None},
+            {"state": TrialState.COMPLETE},
+        ),
+        ({"value": 1.1}, {"value": 1.1}),
+        ({"intermediate_values": {1: 2.3, 3: 2.5}}, {"values": {1: 2.3, 3: 2.5}}),
+        (
+            {
+                "params": {"paramA": 3, "paramB": "bar",},
+                "_distributions": {
+                    "paramA": UniformDistribution(0, 3),
+                    "paramB": CategoricalDistribution(("foo", "bar")),
+                },
+            },
+            {
+                "params": {
+                    "paramA": UniformDistribution(0, 3).to_internal_repr(3),
+                    "paramB": CategoricalDistribution(["foo", "bar"]).to_internal_repr("bar"),
+                },
+                "dists": {
+                    "paramA": UniformDistribution(0, 3),
+                    "paramB": CategoricalDistribution(["foo", "bar"]),
+                },
+            },
+        ),
+        (
+            {"user_attrs": {"attrA": 2.3, "attrB": "foo"}},
+            {"user_attrs": {"attrA": 2.3, "attrB": "foo"}},
+        ),
+        (
+            {"system_attrs": {"attrC": 2.3, "attrB": "bar"}},
+            {"system_attrs": {"attrC": 2.3, "attrB": "bar"}},
+        ),
+    ]  # type: List[Tuple[Dict[str, Any], Dict[str, Any]]]
+
+    for fields_to_modify, kwargs in test_sets:
```

Comment:
IMO, we can make the logic shallow if you use `@pytest.mark.parametrize`. What do you think?

```python
@pytest.mark.parametrize(
    "fields_to_modify, kwargs",
    [
        (
            {"state": TrialState.COMPLETE, "datetime_complete": None},
            {"state": TrialState.COMPLETE},
        ),
        ...
    ],
)
def test_update_trial(fields_to_modify: Dict[str, Any], kwargs: Dict[str, Any]) -> None:
    storage = create_test_storage()
    study_id = storage.create_new_study()

    trial_id = storage.create_new_trial(study_id)
    trial_before_update = storage.get_trial(trial_id)
    ....
```

Reply:
It looks nice and I'll modify the code accordingly.

Reply:
Thanks! I confirmed the change https://github.com/optuna/optuna/pull/1140/commits/740c5961116aa1a6ae8a1637c8a87b9c62a1bc55.